tests/rules/data/syntax-error-in-prepared-statement.php

```diff
@@ -322,6 +322,7 @@ public function errorOnQueryWithoutArgs(\Doctrine\DBAL\Connection $connection)
     public function preparedNamedParamsSubstitution(Connection $connection)
     {
         $connection->preparedQuery('SELECT email FROM ada WHERE email = :param OR email = :parameter', ['param' => 'abc', 'parameter' => 'def']);
+        $connection->preparedQuery('INSERT into ada(adaid, gesperrt, email,freigabe1u1) values(:adaid, :gesperrt, :email, :freigabe1u1)', ['adaid' => 1, 'gesperrt' => 0, 'email' => 'test@github.com', 0]);
```

Comment:
I think the unit test should verify that the insert/update/deletes did not actually modify data in the tables (the statements are rolled back)

Reply:
does the current logic error when passing e.g. a `float` to a parameter expecting a `string`?
we should test that

Reply:
> I think we need a unit test to verify that the insert/update/deletes did not actually modify data in the tables (the statements are rolled back)

not sure where in tests I can get the actual connection - the test cases there are using phpstan rules/return type DI and I wasnt sure how to get the actual connection from there (I was thinking of using QueryReflection::reflector() but thats private)

```php
    public function testInsertRollback()
    {
        $countBefore = $connection->fetchSingle('SELECT count(*) FROM ada');
        assertNotNull($countBefore);

        $connection->query('INSERT into ada(adaid, gesperrt, email, freigabe1u1) values(4, 1, "test@github.com", 0)');

        $countAfter = $connection->fetchSingle('SELECT count(*) FROM ada');
        assertNotNull($countAfter);
        assertEquals($countAfter, $countBefore);
    }
```

I had in mind something like this but these asserts didnt work when I added this code into dibi.php

Reply:
How the test works dependa on the Testclass beeing used. 

DbaInferenceTestCase is for type inference- therefore you use assertType() in the files beeing included.

Other testcases test only a single rule and you need to assert, within the test-class itself, at which line errors are expected for the files beeing included.

Reply:
Maybe we can just test expectating, whether start transaction and rollback was invoked by the test beeing executed?

Reply:
> You might even add a new plain phpunit testcase, which executes the query and tests expectations

but i dont think I can get the db driver from reflector, its all private. and even if i did have the connection in a new plain phpunit test case, how could I test the reflector? Since the reflection/type inference is tested inside the extended phpstan testcases

> Maybe we can just test expectating, whether start transaction and rollback was invoked by the test beeing executed?

you mean to track what commands were executed on the driver and check that there was start transaction / rollback? But normal asserts do not work inside phpstan testcases (i tried adding `assertNotNull(1);` in dibi.php and the tests passed)